GVRf/Sample/gvr-sample/src/org/gearvrf/sample/SampleCubeScript.java

```diff
@@ -160,50 +154,77 @@ public void onInit(GVRContext gvrContext) {
     @Override
     public void onStep() {
         FPSCounter.tick();
-        mFrontBoardObject.getRenderData().getMaterial().setOpacity(1.0f);
-        mFrontBoardObject2.getRenderData().getMaterial().setOpacity(1.0f);
-        mFrontBoardObject3.getRenderData().getMaterial().setOpacity(1.0f);
+        mFrontFace.getRenderData().getMaterial().setOpacity(1.0f);
+        mFrontFace2.getRenderData().getMaterial().setOpacity(1.0f);
+        mFrontFace3.getRenderData().getMaterial().setOpacity(1.0f);
         GVREyePointeeHolder[] eyePointeeHolders = GVRPicker
                 .pickScene(mGVRContext.getMainScene());
         for (GVREyePointeeHolder eyePointeeHolder : eyePointeeHolders) {
-            if (eyePointeeHolder.getOwnerObject().equals(mFrontBoardObject)) {
-                mFrontBoardObject.getRenderData().getMaterial()
-                        .setOpacity(0.5f);
+            if (eyePointeeHolder.getOwnerObject().equals(mFrontFace)) {
+                mFrontFace.getRenderData().getMaterial().setOpacity(0.5f);
             }
-            if (eyePointeeHolder.getOwnerObject().equals(mFrontBoardObject2)) {
-                mFrontBoardObject2.getRenderData().getMaterial()
-                        .setOpacity(0.5f);
+            if (eyePointeeHolder.getOwnerObject().equals(mFrontFace2)) {
+                mFrontFace2.getRenderData().getMaterial().setOpacity(0.5f);
             }
-            if (eyePointeeHolder.getOwnerObject().equals(mFrontBoardObject3)) {
-                mFrontBoardObject3.getRenderData().getMaterial()
-                        .setOpacity(0.5f);
+            if (eyePointeeHolder.getOwnerObject().equals(mFrontFace3)) {
+                mFrontFace3.getRenderData().getMaterial().setOpacity(0.5f);
             }
         }
     }
 
+    private boolean lastScreenshotLeftFinished = true;
```

Comment:
I don't think you want four flags like this - **all** the screenshot APIs are memory and CPU intensive.